pkg/tsdb/prometheus/json_to_dataframes_test.go

```diff
@@ -0,0 +1,120 @@
+package prometheus
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/experimental"
+	"github.com/prometheus/client_golang/api"
+	apiv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMatrixResponses(t *testing.T) {
+	t.Run("parse a simple matrix response", func(t *testing.T) {
+		testScenario(t, "range_simple")
+	})
+
+	t.Run("parse a simple matrix response with value missing steps", func(t *testing.T) {
+		testScenario(t, "range_missing")
+	})
+
+	t.Run("parse a response with Infinity", func(t *testing.T) {
+		testScenario(t, "range_infinity")
+	})
+
+	t.Run("parse a response with NaN", func(t *testing.T) {
+		testScenario(t, "range_nan")
+	})
+}
+
+type mockedRoundTripper struct {
+	responseBytes []byte
+}
+
+func (mockedRT *mockedRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	return &http.Response{
+		StatusCode: http.StatusOK,
+		Body:       io.NopCloser(bytes.NewReader(mockedRT.responseBytes)),
+	}, nil
+}
+
+func makeMockedClient(responseBytes []byte) (apiv1.API, error) {
+	roundTripper := mockedRoundTripper{responseBytes: responseBytes}
+
+	cfg := api.Config{
+		Address:      "http://localhost:9999",
+		RoundTripper: &roundTripper,
+	}
+
+	client, err := api.NewClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	client2 := apiv1.NewAPI(client)
```

Comment:
Nit pick, but I think it would make sense to update the naming because it might seem like we have 2 clients, just to make it clear.

Reply:
aaaaa, i forgot to clean up that code, good catch! will fix it.

Reply:
i changed `client2` to be `api`, this seems how it is named in the prometheus-library.